app/models/simple_inline_text_annotation/entity_type_collection.rb

```diff
@@ -1,8 +1,11 @@
 class SimpleInlineTextAnnotation
   class EntityTypeCollection
     # ENTITY_TYPE_PATTERN matches a pair of square brackets which is followed by a colon and URL.
-    # Example: [Label]: URL
-    ENTITY_TYPE_PATTERN = /^\[([^\]]+)\]:\s+(.*)/
+    # Similar to Markdown, this also matches when there is text enclosed in "" or '' after the URL.
+    # Match example:
+    #  - [Label]: URL
+    #  - [Label]: URL "text"
+    ENTITY_TYPE_PATTERN = /^\[([^\]]+)\]:\s+(\S+)(?:\s+(?:"[^"]*"|'[^']*'))?\s*$/
```

Comment:
```suggestion
    REFERENCE_LINK_PATTERN = /^\[([^\]]+)\]:\s+(\S+)(?:\s+(?:"[^"]*"|'[^']*'))?/
    ENTITY_TYPE_PATTERN = /#{REFERENCE_LINK_PATTERN}\s*$/
```

こんな感じで正規表現の中に別の正規表現を埋め込めるようです。

ENTITY_TYPE_BLOCK_PATTERN と重複部分が共通化でき、部分の意味を表現できます。
読みやすくなりそうです。

Reply:
このように修正してみましたが、いかがでしょうか？
https://github.com/pubannotation/pubannotation/pull/160/commits/c79d8db244824a235538c80a1ceb9319efd06ea9